include/rovio/PoseUpdate.hpp

```diff
@@ -60,8 +60,10 @@ class PoseUpdateMeas: public LWF::State<LWF::VectorElement<3>,LWF::QuaternionEle
  public:
   static constexpr unsigned int _pos = 0;
   static constexpr unsigned int _att = _pos+1;
+  M3D pos_cov_; // Will be used to scale the update covariance according to the measurement
```

Comment:
Ok, it makes sense. It would be a global toggle + the two existing ones for
position and attitude.

I've thought about non-linearities in the innovation covariance inversion
step (`Py_`) and believe that the off-diagonal blocks should be zeroed out
if either position or attitude is set to be unused. This is not a problem
in the current code because `updnoiP_` is diagonal by construction, but it
would be if we allow arbitrary 6x6 covariance matrices. Do you agree?

In other words, the current code relies on the fact that the inverse of a
block-diagonal matrix is still a block-diagonal matrix where the inversion
can be done blockwise. This is not true in the general case we want to
allow here ([reference](
https://en.wikipedia.org/wiki/Block_matrix#Block_matrix_inversion).

On Friday, 5 August 2016, bloesch <notifications@github.com> wrote:

> In include/rovio/PoseUpdate.hpp
> <https://github.com/ethz-asl/rovio/pull/78#discussion_r73651839>:
>
> > @@ -60,8 +60,10 @@ class PoseUpdateMeas: public LWF::State<LWF::VectorElement<3>,LWF::QuaternionEle
> >   public:
> >    static constexpr unsigned int _pos = 0;
> >    static constexpr unsigned int _att = _pos+1;
> > +  M3D pos_cov_; // Will be used to scale the update covariance according to the measurement
>
> The 6x6 matrix would also consider cross-correlations between position and
> attitude. But I see the problem with the potentially invalid data. What
> about making the reading in the callback (RovioNode.hpp) dependent on
> enablePosition_/enableAttitude_?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ethz-asl/rovio/pull/78/files/f472b3bc6a3173bd9d195bfb42d4197e965d1eea#r73651839>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKO4B6i_Kq1kTasQ_-NcOyURB-zP2Rowks5qcuT3gaJpZM4JcIxM>
> .
>